Jetsurvey/app/src/main/java/com/example/compose/jetsurvey/survey/SurveyQuestions.kt

```diff
@@ -19,216 +19,74 @@ package com.example.compose.jetsurvey.survey
-                    options = question.answer.options,
-                    answer = answer as Answer.SingleChoice?,
-                    onAnswerSelected = { answer -> onAnswer(Answer.SingleChoice(answer)) },
-                    modifier = Modifier.fillMaxWidth()
-                )
-                is PossibleAnswer.MultipleChoice -> MultipleChoiceQuestion(
-                    possibleAnswer = question.answer,
-                    answer = answer as Answer.MultipleChoice?,
-                    onAnswerSelected = { newAnswer, selected ->
-                        // create the answer if it doesn't exist or
-                        // update it based on the user's selection
-                        if (answer == null) {
-                            onAnswer(Answer.MultipleChoice(setOf(newAnswer)))
-                        } else {
-                            onAnswer(answer.withAnswerSelected(newAnswer, selected))
-                        }
-                    },
-                    modifier = Modifier.fillMaxWidth()
-                )
-                is PossibleAnswer.Action -> ActionQuestion(
-                    questionId = question.id,
-                    possibleAnswer = question.answer,
-                    answer = answer as Answer.Action?,
-                    onAction = onAction,
-                    modifier = Modifier.fillParentMaxWidth()
-                )
-                is PossibleAnswer.Slider -> SliderQuestion(
-                    possibleAnswer = question.answer,
-                    answer = answer as Answer.Slider?,
-                    onAnswerSelected = { onAnswer(Answer.Slider(it)) },
-                    modifier = Modifier.fillParentMaxWidth()
-                )
-            }
+        Spacer(Modifier.height(32.dp))
+        QuestionTitle(titleResourceId)
+        directionResourceId?.let {
+            Spacer(Modifier.height(18.dp))
+            QuestionDirections(it)
         }
+        Spacer(Modifier.height(18.dp))
+
+        content()
     }
 }
 
 @Composable
-private fun QuestionTitle(@StringRes title: Int) {
-    Row(
+fun QuestionTitle(@StringRes title: Int) {
+    Text(
+        text = stringResource(id = title),
+        style = MaterialTheme.typography.titleMedium,
+        color = MaterialTheme.colorScheme.onSurface.copy(alpha = slightlyDeemphasizedAlpha),
         modifier = Modifier
             .fillMaxWidth()
             .background(
                 color = MaterialTheme.colorScheme.inverseOnSurface,
                 shape = MaterialTheme.shapes.small
             )
-    ) {
-        Text(
-            text = stringResource(id = title),
-            style = MaterialTheme.typography.titleMedium,
-            color = MaterialTheme.colorScheme.onSurface.copy(alpha = slightlyDeemphasizedAlpha),
-            modifier = Modifier
-                .fillMaxWidth()
-                .padding(vertical = 24.dp, horizontal = 16.dp)
-        )
-    }
+            .padding(vertical = 24.dp, horizontal = 16.dp)
```

Comment:
Consider passing this padding from the parent. Conceptually, a QuestionTitle should not know where it's placed in its parent and thus not care about this padding. Having all these paddings defined in the parent makes it easier to reason about how all children relate to each other.

Reply:
I played around with moving this to the parent, but it ends up a little odd. The problem is that the title has a background color that's padded and then the text goes in that. Mentally, I map that background + text as the `QuestionTitle` so the padding is part of that. Another option would be the title knows about the background but the padding is passed in, but that feels like it means the parent knows too much about what the child looks like. I started to think about it from the perspective of "if QuestionTitle is shown elsewhere, should it be with the same background and padding" and I _think_ the answer is yes, so it makes sense to have those defined in the QuestionTitle. What do you think?